### DISCUSSION TITLE
How to use pipeline overrides?

### DISCUSSION THREAD

**Discussion by Pauan** [gfx-rs/wgpu#1748] [Q&A] [answered]
WebGPU (and WGSL) have a concept called [pipeline overrides](https://gpuweb.github.io/gpuweb/wgsl/#pipeline-overridable). They have the same purpose as `#ifdef` preprocessors in GLSL, specialization constants in SPIR-V, and function constants in Metal.

But I can't figure out how to use them with wgpu, I don't see them anywhere in the API.

**Comment by cwfitzgerald** [comment, marked as answer]
They are currently unimplemented, which is why you can't find them! There's an issue outstanding about implementing them.

**Reply by Pauan** [reply]
I had searched for an issue, but I couldn't find anything. Which issue is it?

**Reply by kvark** [reply]
Just filed https://github.com/gfx-rs/wgpu/issues/1762

**Reply by Pauan** [reply]
@kvark Thanks!
